src/tokenizedSubaccounts/CCTSA.sol

```diff
@@ -334,30 +251,37 @@ contract CoveredCallTSA is BaseOnChainSigningTSA {
     if (block.timestamp >= expiry) {
       revert CCT_OptionExpired();
     }
-    uint timeToExpiry = expiry - block.timestamp;
-    if (timeToExpiry < $.ccParams.optionMinTimeToExpiry || timeToExpiry > $.ccParams.optionMaxTimeToExpiry) {
-      revert CCT_OptionExpiryOutOfBounds();
+    (uint callPrice,, uint callDelta) = _getOptionPrice(expiry, strike);
+
+    if (callDelta > $.ccParams.optionMaxDelta) {
+      revert CCT_OptionDeltaTooHigh();
     }
 
-    (uint vol, uint forwardPrice) = _getFeedValues(strike.toUint128(), expiry.toUint64());
+    if (limitPrice <= callPrice) {
+      revert CCT_OptionPriceTooLow();
+    }
+  }
 
-    (uint callPrice,, uint callDelta) = Black76.pricesAndDelta(
+  function _getOptionPrice(uint optionExpiry, uint optionStrike)
+    internal
+    view
+    returns (uint callPrice, uint putPrice, uint callDelta)
+  {
+    CCTSAStorage storage $ = _getCCTSAStorage();
+    uint timeToExpiry = optionExpiry - block.timestamp;
+    if (timeToExpiry < $.ccParams.optionMinTimeToExpiry || timeToExpiry > $.ccParams.optionMaxTimeToExpiry) {
+      revert CCT_OptionExpiryOutOfBounds();
+    }
+    (uint vol, uint forwardPrice) = _getFeedValues(optionStrike.toUint128(), optionExpiry.toUint64());
+    return Black76.pricesAndDelta(
       Black76.Black76Inputs({
         timeToExpirySec: timeToExpiry.toUint64(),
-        volatility: (vol.multiplyDecimal($.ccParams.optionVolSlippageFactor)).toUint128(),
```

Comment:
this shouldn't have been changed/removed. This would actually prevent this contract from working as expected

I'd revert this function back to the original state

Reply:
Great catch. Hmmm I guess this got missed when I moved the functions back and forth between the base class and this class.